### ISSUE TITLE
When I trained my datasets, the loss is always 0

### ISSUE THREAD

**Issue by zhp12343** [Sense-X/Co-DETR#12]
2023-07-13 17:05:51,465 - mmdet - INFO - Epoch [1][450/3061]    lr: 2.000e-05, eta: 1 day, 12:59:23, time: 1.170, data_time: 0.004, memory: 28919, enc_loss_cls: 0.0000, enc_loss_bbox: 0.0000, enc_loss_iou: 0.0000, loss_cls: 0.0000, loss_bbox: 0.0000, loss_iou: 0.0000, d0.loss_cls: 0.0000, d0.loss_bbox: 0.0000, d0.loss_iou: 0.0000, d1.loss_cls: 0.0000, d1.loss_bbox: 0.0000, d1.loss_iou: 0.0000, d2.loss_cls: 0.0000, d2.loss_bbox: 0.0000, d2.loss_iou: 0.0000, d3.loss_cls: 0.0000, d3.loss_bbox: 0.0000, d3.loss_iou: 0.0000, d4.loss_cls: 0.0000, d4.loss_bbox: 0.0000, d4.loss_iou: 0.0000, loss_rpn_cls: 0.0000, loss_rpn_bbox: 0.0000, loss_cls0: 0.0000, acc0: 100.0000, loss_bbox0: 0.0000, loss_cls1: 0.0000, loss_bbox1: 0.0000, loss_centerness1: 0.0000, loss_cls_aux0: 0.0000, loss_bbox_aux0: 0.0000, loss_iou_aux0: 0.0000, d0.loss_cls_aux0: 0.0000, d0.loss_bbox_aux0: 0.0000, d0.loss_iou_aux0: 0.0000, d1.loss_cls_aux0: 0.0000, d1.loss_bbox_aux0: 0.0000, d1.loss_iou_aux0: 0.0000, d2.loss_cls_aux0: 0.0000, d2.loss_bbox_aux0: 0.0000, d2.loss_iou_aux0: 0.0000, d3.loss_cls_aux0: 0.0000, d3.loss_bbox_aux0: 0.0000, d3.loss_iou_aux0: 0.0000, d4.loss_cls_aux0: 0.0000, d4.loss_bbox_aux0: 0.0000, d4.loss_iou_aux0: 0.0000, loss_cls_aux1: 0.0000, loss_bbox_aux1: 0.0000, loss_iou_aux1: 0.0000, d0.loss_cls_aux1: 0.0000, d0.loss_bbox_aux1: 0.0000, d0.loss_iou_aux1: 0.0000, d1.loss_cls_aux1: 0.0000, d1.loss_bbox_aux1: 0.0000, d1.loss_iou_aux1: 0.0000, d2.loss_cls_aux1: 0.0000, d2.loss_bbox_aux1: 0.0000, d2.loss_iou_aux1: 0.0000, d3.loss_cls_aux1: 0.0000, d3.loss_bbox_aux1: 0.0000, d3.loss_iou_aux1: 0.0000, d4.loss_cls_aux1: 0.0000, d4.loss_bbox_aux1: 0.0000, d4.loss_iou_aux1: 0.0000, loss: 0.0000, grad_norm: 0.0003
****

**Comment by zhp12343** [comment]
Could you tell me the reason? I train the model-co_deformable_detr_swin_base_3x_coco.py with the pretrained model that you have given.

**Comment by TempleX98** [comment]
This issue may help: https://github.com/open-mmlab/mmdetection/issues/6911

**Comment by zhp12343** [comment]
> This issue may help: [open-mmlab/mmdetection#6911](https://github.com/open-mmlab/mmdetection/issues/6911)

I have changed the classes.

**Comment by TempleX98** [comment]
Do you specify an area for bbox in your annotations? And you can check the `gt_labels` and `gt_bboxes` in [forward](https://github.com/Sense-X/Co-DETR/blob/27b85c7b830c3db998b8fe384ec4371f979adfde/projects/models/co_detr.py#L117C2-L117C2)

**Comment by zhp12343** [comment]
> 

I have trained mmdetection-deformable_detr which is used the gt json.
I want to know if it is because the model have filtered images too small,such as 10 pixes?

**Comment by TempleX98** [comment]
Yes, code is [here](https://github.com/Sense-X/Co-DETR/blob/27b85c7b830c3db998b8fe384ec4371f979adfde/mmdet/datasets/custom.py#L178C9-L178C21) and the min_size is 32

**Comment by zhp12343** [comment]
> Yes, code is [here](https://github.com/Sense-X/Co-DETR/blob/27b85c7b830c3db998b8fe384ec4371f979adfde/mmdet/datasets/coco.py#L121) and the min_size is 32

I changed the min_size is 0. However, some of the loss is still 0.
2023-07-13 18:04:37,344 - mmdet - INFO - Epoch [1][50/3060]     lr: 1.000e-06, eta: 1 day, 16:35:20, time: 1.327, data_time: 0.052, memory: 23981, enc_loss_cls: 0.7951, enc_loss_bbox: 0.0000, enc_loss_iou: 0.0000, loss_cls: 0.0029, loss_bbox: 0.0000, loss_iou: 0.0000, d0.loss_cls: 0.0105, d0.loss_bbox: 0.0000, d0.loss_iou: 0.0000, d1.loss_cls: 0.0058, d1.loss_bbox: 0.0000, d1.loss_iou: 0.0000, d2.loss_cls: 0.0052, d2.loss_bbox: 0.0000, d2.loss_iou: 0.0000, d3.loss_cls: 0.0067, d3.loss_bbox: 0.0000, d3.loss_iou: 0.0000, d4.loss_cls: 0.0024, d4.loss_bbox: 0.0000, d4.loss_iou: 0.0000, loss_rpn_cls: 7.2219, loss_rpn_bbox: 0.0000, loss_cls0: 4.1905, acc0: 96.6528, loss_bbox0: 0.0000, loss_cls1: 0.5143, loss_bbox1: 0.0000, loss_centerness1: 0.0000, loss_cls_aux0: 0.0001, loss_bbox_aux0: 0.0000, loss_iou_aux0: 0.0000, d0.loss_cls_aux0: 0.0001, d0.loss_bbox_aux0: 0.0000, d0.loss_iou_aux0: 0.0000, d1.loss_cls_aux0: 0.0001, d1.loss_bbox_aux0: 0.0000, d1.loss_iou_aux0: 0.0000, d2.loss_cls_aux0: 0.0002, d2.loss_bbox_aux0: 0.0000, d2.loss_iou_aux0: 0.0000, d3.loss_cls_aux0: 0.0002, d3.loss_bbox_aux0: 0.0000, d3.loss_iou_aux0: 0.0000, d4.loss_cls_aux0: 0.0001, d4.loss_bbox_aux0: 0.0000, d4.loss_iou_aux0: 0.0000, loss_cls_aux1: 0.0001, loss_bbox_aux1: 0.0000, loss_iou_aux1: 0.0000, d0.loss_cls_aux1: 0.0003, d0.loss_bbox_aux1: 0.0000, d0.loss_iou_aux1: 0.0000, d1.loss_cls_aux1: 0.0002, d1.loss_bbox_aux1: 0.0000, d1.loss_iou_aux1: 0.0000, d2.loss_cls_aux1: 0.0002, d2.loss_bbox_aux1: 0.0000, d2.loss_iou_aux1: 0.0000, d3.loss_cls_aux1: 0.0002, d3.loss_bbox_aux1: 0.0000, d3.loss_iou_aux1: 0.0000, d4.loss_cls_aux1: 0.0001, d4.loss_bbox_aux1: 0.0000, d4.loss_iou_aux1: 0.0000, loss: 12.7571, grad_norm: 89.1607

**Comment by zhp12343** [comment]
2023-07-13 18:07:40,035 - mmdet - INFO - Epoch [1][200/3060]    lr: 1.000e-06, eta: 1 day, 14:02:02, time: 1.187, data_time: 0.004, memory: 26302, enc_loss_cls: 0.1348, enc_loss_bbox: 0.0000, enc_loss_iou: 0.0000, loss_cls: 0.0000, loss_bbox: 0.0000, loss_iou: 0.0000, d0.loss_cls: 0.0004, d0.loss_bbox: 0.0000, d0.loss_iou: 0.0000, d1.loss_cls: 0.0001, d1.loss_bbox: 0.0000, d1.loss_iou: 0.0000, d2.loss_cls: 0.0000, d2.loss_bbox: 0.0000, d2.loss_iou: 0.0000, d3.loss_cls: 0.0000, d3.loss_bbox: 0.0000, d3.loss_iou: 0.0000, d4.loss_cls: 0.0000, d4.loss_bbox: 0.0000, d4.loss_iou: 0.0000, loss_rpn_cls: 0.9271, loss_rpn_bbox: 0.0000, loss_cls0: 0.0007, acc0: 100.0000, loss_bbox0: 0.0000, loss_cls1: 0.0043, loss_bbox1: 0.0000, loss_centerness1: 0.0000, loss_cls_aux0: 0.0000, loss_bbox_aux0: 0.0000, loss_iou_aux0: 0.0000, d0.loss_cls_aux0: 0.0000, d0.loss_bbox_aux0: 0.0000, d0.loss_iou_aux0: 0.0000, d1.loss_cls_aux0: 0.0000, d1.loss_bbox_aux0: 0.0000, d1.loss_iou_aux0: 0.0000, d2.loss_cls_aux0: 0.0000, d2.loss_bbox_aux0: 0.0000, d2.loss_iou_aux0: 0.0000, d3.loss_cls_aux0: 0.0000, d3.loss_bbox_aux0: 0.0000, d3.loss_iou_aux0: 0.0000, d4.loss_cls_aux0: 0.0000, d4.loss_bbox_aux0: 0.0000, d4.loss_iou_aux0: 0.0000, loss_cls_aux1: 0.0000, loss_bbox_aux1: 0.0000, loss_iou_aux1: 0.0000, d0.loss_cls_aux1: 0.0000, d0.loss_bbox_aux1: 0.0000, d0.loss_iou_aux1: 0.0000, d1.loss_cls_aux1: 0.0000, d1.loss_bbox_aux1: 0.0000, d1.loss_iou_aux1: 0.0000, d2.loss_cls_aux1: 0.0000, d2.loss_bbox_aux1: 0.0000, d2.loss_iou_aux1: 0.0000, d3.loss_cls_aux1: 0.0000, d3.loss_bbox_aux1: 0.0000, d3.loss_iou_aux1: 0.0000, d4.loss_cls_aux1: 0.0000, d4.loss_bbox_aux1: 0.0000, d4.loss_iou_aux1: 0.0000, loss: 1.0675, grad_norm: 15.5428

**Comment by TempleX98** [comment]
It seems the `gt_bboxes` are filtered or not loaded

**Comment by zhp12343** [comment]
> It seems the `gt_bboxes` are filtered or not loaded

How to solve it?

**Comment by TempleX98** [comment]
Check if `img_metas`, `gt_bboxes` and `gt_labels` in [forward](https://github.com/Sense-X/Co-DETR/blob/27b85c7b830c3db998b8fe384ec4371f979adfde/projects/models/co_detr.py#L117C2-L117C2)  are consistent with your annotations

**Comment by zhp12343** [comment]
I use this ‘/mmyolo-main/tools/dataset_converters/yolo2coco.py’ generated the gt json file

**Comment by zhp12343** [comment]
Can you tell me your email, I'll send the json over, can you check it for me

**Comment by zhp12343** [comment]
I found the reason! I modified classes=('a','b','c') as per customize_dataset.md，but I found that defining classes this way, I only got one class,it is ('a','b','c) as a whole.
So I modified classes = 'a','b','c', the loss returns to normal.
I'm guessing there is an error in passing the classes parameter in your code！Please check it!

**Comment by TempleX98** [comment]
I will check it
